init/bootstrap.go

```diff
@@ -43,10 +46,18 @@ outer:
 			continue
 		}
 
-		for _, b := range buffer {
-			if b != 0 {
-				log.Infof("%s not empty", dev)
-				continue outer
+		boot2docker := false
+
+		if strings.HasPrefix(string(buffer), boot2dockerMagic) {
```

Comment:
Okay, so I'm wrong it does copy the bytes.  There is specific magic in golang when going between []byte and string.  Good to know.